ludwig/config_validation/checks.py

```diff
@@ -639,6 +640,29 @@ def check_qlora_requirements(config: "ModelConfig") -> None:  # noqa: F821
         raise ConfigValidationError("Fine-tuning and LLM with quantization requires using the 'lora' adapter")
 
 
+@register_config_check
+def check_qlora_merge_and_unload_compatibility(config: "ModelConfig") -> None:  # noqa: F821
+    """Checks that model.merge_and_unload() is supported by underlying model.save_pretrained() when merging QLoRA
+    layers."""
+    if config.model_type != MODEL_LLM or config.trainer.type == "none":
+        return
+
+    if not (
+        config.adapter
+        and config.adapter.type == "lora"
+        and config.adapter.postprocessor
+        and config.adapter.postprocessor.merge_adapter_into_base_model
+        and config.quantization
+    ):
+        return
+
+    if config.quantization.bits < MIN_QUANTIZATION_BITS_FOR_MERGE_AND_UNLOAD:
+        raise ConfigValidationError(
+            f"""This operation will entail merging LoRA layers on a {config.quantization.bits}-bit \
+quantized model.  Calling "save_pretrained()" on that model is currently unsupported."""
+        )
```

Comment:
Maybe we can use this as an opportunity to let users know how to rectify the issue - perhaps adding something like "If you want to merge the LoRA adapter weights into the base model, you need to do 8-bit quantization or do non-quantized based training by removing the quantization section from your Ludwig config".

Reply:
@arnavgarg1 This has been incorporated.  Thanks!